lib/mini_profiler/snapshots_transporter.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class ::Rack::MiniProfiler::SnapshotsTransporter
+  class << self
+    def transport(snapshot)
+      @transporter ||= self.new(Rack::MiniProfiler.config)
+      @transporter.ship(snapshot)
+    end
+  end
+
+  attr_reader :buffer
+  attr_accessor :max_buffer_size
+
+  def initialize(config)
+    @uri = URI(config.snapshots_transport_destination_url)
+    @auth_key = config.snapshots_transport_auth_key
+    @thread = nil
+    @thread_mutex = Mutex.new
+    @buffer = []
+    @buffer_mutex = Mutex.new
+    @max_buffer_size = 100
+    @testing = false
+  end
+
+  def ship(snapshot)
+    @buffer_mutex.synchronize do
+      @buffer << snapshot
+      @buffer.shift if @buffer.size > @max_buffer_size
+    end
+    @thread_mutex.synchronize { start_thread }
+  end
+
+  def flush_buffer
+    if @buffer.size > 0
+      buffer_content = @buffer.dup
+      request = Net::HTTP::Post.new(
+        @uri,
+        'Content-Type' => 'application/json',
+        'Mini-Profiler-Transport-Auth' => @auth_key
+      )
+      request.body = { snapshots: buffer_content }.to_json
+      http = Net::HTTP.new(@uri.hostname, @uri.port)
+      http.use_ssl = @uri.scheme == 'https'
+      res = http.request(request)
+      if res.code.to_i == 200
```

Comment:
we should probably at least expose some top level metrics here under mini profiler namespace for 

- Number of things we shipped successfully 
- Number of 200 responses
- Number of unknown errors  

attr_readers on this class is probably good enough then we can dig this out using prometheus which is hand for debugging issues.